azure-keyvault/src/test/java/com/microsoft/azure/keyvault/test/KeyVaultClientIntegrationTestBase.java

```diff
@@ -1,161 +1,325 @@
-    }
-
-    private static AuthenticationResult getAccessToken(String authorization, String resource) throws Exception {
-
-        String clientId = System.getenv("arm.clientid");
-    
-        if (clientId == null) {
-            throw new Exception("Please inform arm.clientid in the environment settings.");
-        }
-
-        String clientKey = System.getenv("arm.clientkey");
-        String username = System.getenv("arm.username");
-        String password = System.getenv("arm.password");
-        
-        AuthenticationResult result = null;
-        ExecutorService service = null;
-        try {
-            service = Executors.newFixedThreadPool(1);
-            AuthenticationContext context = new AuthenticationContext(authorization, false, service);
-
-            Future<AuthenticationResult> future = null;
-
-            if (clientKey != null && password == null) {
-                ClientCredential credentials = new ClientCredential(clientId, clientKey);
-                future = context.acquireToken(resource, credentials, null);
-            }
-
-            if (password != null && clientKey == null) {
-                future = context.acquireToken(resource, clientId, username, password, null);
-            }
-
-            if (future == null) {
-                throw new Exception("Missing or ambiguous credentials - please inform exactly one of arm.clientkey or arm.password in the environment settings.");
-            }
-
-            result = future.get();
-        } finally {
-            service.shutdown();
-        }
-
-        if (result == null) {
-            throw new RuntimeException("authentication result was null");
-        }
-        return result;
-    }
-
-    protected static void compareAttributes(Attributes expectedAttributes, Attributes actualAttribute) {
-        if(expectedAttributes != null) {
-            Assert.assertEquals(expectedAttributes.enabled(), actualAttribute.enabled());
-            Assert.assertEquals(expectedAttributes.expires(), actualAttribute.expires());
-            Assert.assertEquals(expectedAttributes.notBefore(), actualAttribute.notBefore());
-        }
-    }
-    
-    protected static ObjectWriter jsonWriter;
-    protected static ObjectReader jsonReader;
-
-    @BeforeClass
-    public static void setup() throws Exception {
-        createKeyVaultClient();
-        jsonWriter = new ObjectMapper().writer().withDefaultPrettyPrinter();
-        jsonReader = new ObjectMapper().reader();
-    }
-
-    @AfterClass
-    public static void cleanup() throws Exception {
-    }
-
-    @Before
-    public void beforeTest() throws Exception {
-        //setupTest(getClass().getSimpleName() + "-" + "??");
-    }
-
-    @After
-    public void afterTest() throws Exception {
-        //resetTest(getClass().getSimpleName() + "-" + "??");
-    }
-
-}
\ No newline at end of file
+	protected RestClient buildRestClient(RestClient.Builder builder, boolean isMocked) {
+		return builder.build();
```

Comment:
the param isMocked doesn't appear to be used here, does this method require this signature for some reason?